File: clients/python/sdk/models/thumbnail.py
from __future__ import annotations
from kiota_abstractions.serialization import AdditionalDataHolder, Parsable, ParseNode, SerializationWriter
from typing import Any, Callable, Dict, List, Optional, Union

from . import thumnail_version

class Thumbnail(AdditionalDataHolder, Parsable):
    @property
    def additional_data(self,) -> Dict[str, Any]:
        """
        Gets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
        Returns: Dict[str, Any]
        """
        return self._additional_data
    
    @additional_data.setter
    def additional_data(self,value: Dict[str, Any]) -> None:
        """
        Sets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
        Args:
            value: Value to set for the AdditionalData property.
        """
        self._additional_data = value
    
    @property
    def blurhash(self,) -> Optional[str]:
        """
        Gets the blurhash property value. The blurhash property
        Returns: Optional[str]
        """
        return self._blurhash
    
    @blurhash.setter
    def blurhash(self,value: Optional[str] = None) -> None:
        """
        Sets the blurhash property value. The blurhash property
        Args:
            value: Value to set for the blurhash property.
        """
        self._blurhash = value
    
    def __init__(self,) -> None:
        """
        Instantiates a new Thumbnail and sets the default values.
        """
        # Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
        self._additional_data: Dict[str, Any] = {}

        # The blurhash property
        self._blurhash: Optional[str] = None
        # The url property
        self._url: Optional[str] = None
        # The versions property
        self._versions: Optional[thumnail_version.ThumnailVersion] = None
    
    @staticmethod
    def create_from_discriminator_value(parse_node: Optional[ParseNode] = None) -> Thumbnail:
        """
        Creates a new instance of the appropriate class based on discriminator value
        Args:
            parseNode: The parse node to use to read the discriminator value and create the object
        Returns: Thumbnail
        """
        if parse_node is None:
            raise Exception("parse_node cannot be undefined")
        return Thumbnail()
    
    def get_field_deserializers(self,) -> Dict[str, Callable[[ParseNode], None]]:
        """
        The deserialization information for the current model
        Returns: Dict[str, Callable[[ParseNode], None]]
        """
        fields = {
            "blurhash": lambda n : setattr(self, 'blurhash', n.get_str_value()),
            "url": lambda n : setattr(self, 'url', n.get_str_value()),
            "versions": lambda n : setattr(self, 'versions', n.get_object_value(thumnail_version.ThumnailVersion)),
        }
        return fields
    
    def serialize(self,writer: SerializationWriter) -> None:
        """
        Serializes information the current object
        Args:
            writer: Serialization writer to use to serialize this model
        """
        if writer is None:
            raise Exception("writer cannot be undefined")
        writer.write_str_value("blurhash", self.blurhash)
        writer.write_str_value("url", self.url)
        writer.write_object_value("versions", self.versions)
        writer.write_additional_data_value(self.additional_data)
    
    @property
    def url(self,) -> Optional[str]:
        """
        Gets the url property value. The url property
        Returns: Optional[str]
        """
        return self._url
    
    @url.setter
    def url(self,value: Optional[str] = None) -> None:
        """
        Sets the url property value. The url property
        Args:
            value: Value to set for the url property.
        """
        self._url = value
    
    @property
    def versions(self,) -> Optional[thumnail_version.ThumnailVersion]:
        """
        Gets the versions property value. The versions property
        Returns: Optional[thumnail_version.ThumnailVersion]
        """
        return self._versions
    
    @versions.setter
    def versions(self,value: Optional[thumnail_version.ThumnailVersion] = None) -> None:
        """
        Sets the versions property value. The versions property
        Args:
            value: Value to set for the versions property.
        """
        self._versions = value
    

File: clients/node/sdk/models/thumbnail.ts
import {createThumnailVersionFromDiscriminatorValue} from './createThumnailVersionFromDiscriminatorValue';
import {ThumnailVersion} from './index';
import {AdditionalDataHolder, Parsable, ParseNode, SerializationWriter} from '@microsoft/kiota-abstractions';

export class Thumbnail implements AdditionalDataHolder, Parsable {
    /** Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well. */
    private _additionalData: Record<string, unknown>;
    /** The blurhash property */
    private _blurhash?: string | undefined;
    /** The url property */
    private _url?: string | undefined;
    /** The versions property */
    private _versions?: ThumnailVersion | undefined;
    /**
     * Gets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @returns a Record<string, unknown>
     */
    public get additionalData() {
        return this._additionalData;
    };
    /**
     * Sets the additionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @param value Value to set for the AdditionalData property.
     */
    public set additionalData(value: Record<string, unknown>) {
        this._additionalData = value;
    };
    /**
     * Gets the blurhash property value. The blurhash property
     * @returns a string
     */
    public get blurhash() {
        return this._blurhash;
    };
    /**
     * Sets the blurhash property value. The blurhash property
     * @param value Value to set for the blurhash property.
     */
    public set blurhash(value: string | undefined) {
        this._blurhash = value;
    };
    /**
     * Instantiates a new Thumbnail and sets the default values.
     */
    public constructor() {
        this._additionalData = {};
    };
    /**
     * The deserialization information for the current model
     * @returns a Record<string, (node: ParseNode) => void>
     */
    public getFieldDeserializers() : Record<string, (node: ParseNode) => void> {
        return {
            "blurhash": n => { this.blurhash = n.getStringValue(); },
            "url": n => { this.url = n.getStringValue(); },
            "versions": n => { this.versions = n.getObjectValue<ThumnailVersion>(createThumnailVersionFromDiscriminatorValue); },
        };
    };
    /**
     * Serializes information the current object
     * @param writer Serialization writer to use to serialize this model
     */
    public serialize(writer: SerializationWriter) : void {
        if(!writer) throw new Error("writer cannot be undefined");
        writer.writeStringValue("blurhash", this.blurhash);
        writer.writeStringValue("url", this.url);
        writer.writeObjectValue<ThumnailVersion>("versions", this.versions);
        writer.writeAdditionalData(this.additionalData);
    };
    /**
     * Gets the url property value. The url property
     * @returns a string
     */
    public get url() {
        return this._url;
    };
    /**
     * Sets the url property value. The url property
     * @param value Value to set for the url property.
     */
    public set url(value: string | undefined) {
        this._url = value;
    };
    /**
     * Gets the versions property value. The versions property
     * @returns a ThumnailVersion
     */
    public get versions() {
        return this._versions;
    };
    /**
     * Sets the versions property value. The versions property
     * @param value Value to set for the versions property.
     */
    public set versions(value: ThumnailVersion | undefined) {
        this._versions = value;
    };
}
